tchannel/peer_heap.py

```diff
@@ -80,3 +80,12 @@ def peek_peer(self):
             return None
 
         return self.peers[0]
+
+    def remove_peer(self, peer):
+        """Remove the peer from the heap.
+        If the peer's index out of range[0, peer'size), the remove function
+        will do nothing.
+        """
+        if 0 <= peer.index < self.size():
+            p = heap.remove(self, peer.index)
+            assert p == peer
```

Comment:
Also, this assertion is coming after you've already removed the peer. So if it fails, you'd have removed the wrong peer anyway.